_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -148,39 +208,73 @@ Label is included in accessibility tree, but not visible, and does not describe
 <input aria-labelledby="label_fname" type="text" name="fname"/>
 ```
 
-### Inapplicable
+#### Failed Example 6
 
-#### Inapplicable example 1
+Accessible name created through `aria-label` does not describes the purpose of the associated element.
 
-`Label` that is neither visible to users, nor included in the accessibility tree.
+```html
+<input aria-label="Menu" id="fname" type="text" name="fname"/>
+```
+
+#### Failed Example 7
+
+The `label` in itself does not describe the purpose of the form field. While the context can be relied upon to differentiate form fields from each other, it cannot be relied upon for describing the entire purpose of the form field.
 
 ```html
-<label for="fname" style:"display:none;">First name:</label>
-<input id="fname" type="text" name="fname"/>
+<h2>Name</h2>
+<label>Fill in: <input id="name" type="text" name="name"/></label>
+````
+
+#### Failed Example 8
+
+While Expectation 1 is met, Expectation 2 fails because the descriptive label coded with the `label` element through the accessible name calculation is overwritten by an `aria-label` that is not descriptive.
```

Comment:
I'm not sure I would fail this one. While the accessible name is "Fill in:", you get immediately before in the accessibility tree "First name:". 
But, according to the expectation is fails, which means I don't know if we need to work on the expectation. Opinions, anyone?

Reply:
I would fail this Carlos. The accessible name should be descriptive without context in this case I would say.

Reply:
I agree with @ShadowBB. According to the expectation the context can only be used to differentiate similar fields from each other, not to provide the whole description.

Do others have an opinion about this? (e.g. @WilcoFiers, @audreymaniez, @EmmaJP ?)

Reply:
I agree too with @ShadowBB, I would fail this one.